Sources/SourceKitLSP/Swift/MacroExpansion.swift

```diff
@@ -74,28 +74,28 @@ extension SwiftLanguageService {
     var macroExpansionReferenceDocumentURLs: [ReferenceDocumentURL] = []
     for macroEdit in expansion.edits {
       if let bufferName = macroEdit.bufferName {
-        let macroExpansionReferenceDocumentURLData = if let referenceDocumentURL, case let .macroExpansion(referenceDocumentURLData) = referenceDocumentURL {
-          ReferenceDocumentURL.macroExpansion(
-            MacroExpansionReferenceDocumentURLData(
-              macroExpansionEditRange: macroEdit.range,
-              primaryFileURL: primaryFileURL,
-              sourceFileURL: expandMacroCommand.textDocument.uri.arbitrarySchemeURL, // correct for regular
-              selectionRange: expandMacroCommand.positionRange,
-              bufferName: bufferName
+        let macroExpansionReferenceDocumentURLData =
+          if let referenceDocumentURL, case .macroExpansion(_) = referenceDocumentURL {
```

Comment:
I think a `switch` would be better than an `if` here so that you need to consider adding a new case when extending `ReferenceDocumentURL`.